beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/scenario/TestCases.kt

```diff
@@ -0,0 +1,206 @@
+package com.technocreatives.beckon.mesh.scenario
+
+import arrow.core.*
+import arrow.core.continuations.either
+import arrow.fx.coroutines.parZip
+import arrow.typeclasses.Semigroup
+import com.technocreatives.beckon.mesh.BeckonMesh
+import com.technocreatives.beckon.mesh.data.GroupAddress
+import com.technocreatives.beckon.mesh.data.ProxyFilterMessage
+import com.technocreatives.beckon.mesh.data.PublishableAddress
+import com.technocreatives.beckon.mesh.data.UnicastAddress
+import com.technocreatives.beckon.mesh.message.SendVendorModelMessage
+import com.technocreatives.beckon.mesh.message.sendVendorModelMessage
+import com.technocreatives.beckon.mesh.message.sendVendorModelMessageAck
+import com.technocreatives.beckon.mesh.state.Connected
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.withTimeoutOrNull
+import no.nordicsemi.android.mesh.transport.VendorModelMessageStatus
+import timber.log.Timber
+import kotlin.experimental.ExperimentalTypeInference
+import kotlin.math.roundToInt
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+sealed interface Test {
+
+    data class VendorMessageAck(
+        val address: PublishableAddress,
+        val message: SendVendorModelMessage,
+        val responseOpCode: Int,
+        val assert: Assertion<VendorModelMessageStatus>
+    ) : Test
+
+    // for reliability / success-rate
+    data class SuccessRate(
+        val address: GroupAddress,
+        val message: SendVendorModelMessage,
+        val expected: ExpectedSuccessRate,
+    ) : Test
+
+    data class SingleVendorMessage(
+        val address: UnicastAddress,
+        val message: SendVendorModelMessage,
+        val assert: Assertion<ProxyFilterMessage>,
+    ) : Test
+
+    data class AssertOnly(
+        val expectedSize: Int,
+        val assert: Assertion<List<ProxyFilterMessage>>
+    ) : Test
+
+    data class MultipleVendorMessage(
```

Comment:
This is not `MultipleVendorMessage` anymore. Should we move it to `MultipleTests`?